packages/dd-trace/src/log_writer.js

```diff
@@ -0,0 +1,101 @@
+'use strict'
+
+const { storage } = require('../../datadog-core')
+const { Level, subscribe, unsubscribe } = require('./log_channels')
+
+const _defaultLogger = {
+  debug: msg => console.debug(msg), /* eslint-disable-line no-console */
+  info: msg => console.info(msg), /* eslint-disable-line no-console */
+  warn: msg => console.warn(msg), /* eslint-disable-line no-console */
+  error: msg => console.error(msg) /* eslint-disable-line no-console */
+}
+const _listeners = {
+  [Level.Debug]: onDebug,
+  [Level.Info]: onInfo,
+  [Level.Warn]: onWarn,
+  [Level.Error]: onError
+}
+
+let _enabled = false
+let _logger = _defaultLogger
+
+function processMsg (msg) {
```

Comment:
Is this emitter or receiver responsibility? I moved here but it could be in log.js

Reply:
imo, it is responsibility of emitter. If we want to add more subscribers, they will be implement exactly the same code to get the message.

Reply:
May be a subscriber doesn't execute the function to get the message and calls `toString` to get the function body... :P

Reply:
Moved